cf-serverd/cf-serverd-functions.c

```diff
@@ -540,14 +557,99 @@ static void PrepareServer(int sd)
     WritePID("cf-serverd.pid"); /* Arranges for cleanup() to tidy it away */
 }
 
+/* Graceful Stop
+ * This runs a new main process that will die and that init can restart (as systemd can do).
+ * This this can prevent systemd from kill us it there is a problem.
+ * But this makes it possible to finish kandling connections while systemd tries to restart us.
+ * If there is no systemd make sure alternative init restarts us. */
```

Comment:
What's the point of doing all the magic below without systemd?

Reply:
Iif you had another init than systemd that restarts cf-serverd automatically, you could do the same thing, you just need to tell it the main process has changed. 
Depending on the init, this could be done by signaling or by killing the original process instead of the grand-child

Reply:
Sure. But my question really was -- Should this code be guarded by an `#ifdef` making sure it only runs **with systemd**? Because with other init systems, I don't see a reason to do the forking exercise at this point.

Reply:
we could mode the #if HAVE_SYSTEMD_SD_DAEMON_H at the beginning of the function

Reply:
I'd prefer doing in the place where the function is called. Or, actually, both places. No need to have dead code.

Reply:
like this ?